align/schema/constraint.py

```diff
@@ -790,6 +793,7 @@ class GroundPorts(SoftConstraint):
         }
     '''
     ports: List[str]
+    propagate: bool = True
```

Comment:
@kkunal1408 , this is a global switch which turns on/off the propagation for all instances. What if we want few instances to auto-propagate but few other not to? Instead, we can remove this global switch and assume propagation unless a subcircuit defines these ports explicitly.

Reply:
This switch works at a hierarchy level. By default it's true and thus can auto-propagate. Users can turn it off by setting this off for any hierarchy they do not want it to propagate.

Reply:
Let me explain by an example: Suppose there are 7 subcircuits under a top level hierarchy (similar to the bias generator example) and we want to turn propagation off only for one of these subcircuits.
In this approach you are proposing, we have to turn off globally and define power ports for all subcircuits. In other words, when you turn off global, it forces user to specify power ports for all subcircuits in that hierarchy.
In the approach I am proposing, we only need to define power ports for the top-level and the subcircuit that we want to stop propagation.

Reply:
In this case, setting propagate: true for the top and defining power ports for the subcircuit we want and setting propagate: false would do the job.

Reply:
I am not following for which constraint propagate should be set. It also sounds like you are overloading propagate with inherit. Could you please write an example?

Reply:
We need to set propagate: false only for power_cell in this test_case. Modified the test_case to a minimal example. 
Added some more code to control propagation from the top. will add a small test for that.

Reply:
Should not we call this inherit rather than propagate then?

Reply:
The functionality is similar. The keyword is used to synchronize with other propagate constraints.
A small difference from inheritance is that the global nets of a child-hierarchy can be a subset of a parent-hierarchy and they can have different names.